android/app/pom.xml

```diff
@@ -114,7 +114,7 @@
         1. Use src/test/java for JUnit tests that don't call any Android APIs (directly or transitively)
         2. Use src/main/java for JUnit tests that call Android APIs  
         -->
-        <sourceDirectory>src</sourceDirectory>        
+        <sourceDirectory>src/main/java</sourceDirectory>        
```

Comment:
Maybe Eclipse is more strict with the Maven declarations?
I just tested it, it works fine without it at me too.